### DISCUSSION TITLE
can't stop listen 0.0.0.0 for api

### DISCUSSION THREAD

**Discussion by leagueofsoups** [rancher/rke2#4162] [Q&A] [answered]
**Environmental Info:**
RKE2 Version: 
rke2 version v1.27.1-rc1+rke2r1 (9ce3825ec005872a0873307844ba99a2bdd4ee19)
go version go1.20.3 X:boringcrypto

Node(s) CPU architecture, OS, and Version: 
Linux control-1 5.4.0-135-generic #152-Ubuntu SMP Wed Nov 23 20:19:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
1 server

**Describe the bug:**
How I can stop listen 0.0.0.0 by kubeapi? Static pod manifest has localhost hardcode inside probe.
https://github.com/rancher/rke2/blob/0a1f395b79cfe8bdc41b6b69f5430f16c8654eb1/pkg/podexecutor/staticpod.go#L320
https://github.com/rancher/rke2/blob/0a1f395b79cfe8bdc41b6b69f5430f16c8654eb1/pkg/podexecutor/staticpod.go#L329

**Steps To Reproduce:**
Add to your config this fragment:
```
kube-apiserver-arg:
- bind-address=<not 127.0.0.1 ipv4_addr>
bind-address: <not 127.0.0.1 ipv4_addr>
```
Now, probe not working.
* when addresses not equal kubeconfig will contain bad server parameter

**Expected behavior:**
probe ok

**Actual behavior:**
probe not working

**Comment by brandond** [comment, marked as answer]
Binding the apiserver to a specific interface is not currently supported by RKE2. We expect it to be available on both localhost, and the advertised IP, but the bind-address flag does not currently allow for that.
